src/main.rs

```diff
@@ -196,7 +196,11 @@ fn set_env(cx: &Context, env: &mut dyn EnvTarget, IsNextest(is_nextest): IsNexte
                 flags.push("codegen-units=1");
             }
         }
-        if cx.args.cov.branch {
+        if cx.args.cov.mcdc {
+            // Tracking issue: https://github.com/rust-lang/rust/issues/124144
+            flags.push("-Z");
+            flags.push("coverage-options=mcdc");
+        } else if cx.args.cov.branch {
```

Comment:
If both --mcdc and --branch are specified, it would be nice if we could catch those cases and make an error or warning instead of silently letting mcdc win here.

Reply:
I added such a warning in `src/context.rs`, and rebased the PR.